Documentation/guides/MSBuildBestPractices.md

```diff
@@ -32,6 +32,77 @@ properties, targets, etc., as we would likely need to support them
 into oblivion. However, we might choose to "safely" deprecate them
 in a way that makes sense.
 
+## Item Group Transforms
+
+MSBuild has a widely used feature where you can create a one-to-one
+mapping from an `<ItemGroup/>` to a new `<ItemGroup/>`. The syntax for
+this looks like:
+
+```xml
+<ItemGroup>
+  <_DestinationFiles Include="@(_SourceFiles->'$(SomeDirectory)%(Filename)%(Extension)')" />
+</ItemGroup>
+```
+
+This takes a list of files, and creates a desired destination path for
+each file in `$(SomeDirectory)`. The `%(Filename)` and `%(Extension)`
```

Comment:
Would probably be worth linking to the [**MSBuild well-known item metadata** page](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-well-known-item-metadata?view=vs-2017), which *documents* `%(Filename)`.